coalib/results/SourceRange.py

```diff
@@ -10,7 +10,10 @@ class SourceRange(TextRange):
     @enforce_signature
     def __init__(self,
                  start: SourcePosition,
-                 end: (SourcePosition, None)=None):
+                 end: (SourcePosition, None)=None,
+                 text=None,
+                 start_position=None,
+                 end_position=None):
```

Comment:
We need to do assertions in the function to make sure `start_position` and `start` are not given together and so on. Otherwise it's confusing.